src/datapilot/core/platforms/dbt/insights/checks/check_column_name_contract.py

```diff
@@ -0,0 +1,97 @@
+    )
+
+    def __init__(self, catalog_wrapper: BaseCatalogWrapper, *args, **kwargs):
+        self.catalog = catalog_wrapper
+        super().__init__(*args, **kwargs)
+
+    def generate(self, *args, **kwargs) -> List[DBTModelInsightResponse]:
+        self.pattern = get_contract_regex_configuration(self.config)
+        self.dtypes = get_dtypes_configuration(self.config)
+        self.dtypes = self.dtypes if self.dtypes else []
+        insights = []
+        for node_id, node in self.nodes.items():
+            if self.should_skip_model(node_id):
+                self.logger.debug(f"Skipping model {node_id} as it is not enabled for selected models")
+                continue
+            if node.resource_type == AltimateResourceType.model:
+                columns = self._get_columns_with_contract_violation(node_id)
+                if columns:
+                    insights.append(
+                        DBTModelInsightResponse(
+                            unique_id=node_id,
+                            package_name=node.package_name,
+                            path=node.original_file_path,
+                            original_file_path=node.original_file_path,
+                            insight=self._build_failure_result(node_id, columns),
+                            severity=get_severity(self.config, self.ALIAS, self.DEFAULT_SEVERITY),
+                        )
+                    )
+        return insights
+
+    def _build_failure_result(self, model_unique_id: str, columns: Sequence[str]) -> DBTInsightResult:
+        failure_message = self.FAILURE_MESSAGE.format(
+            columns=numbered_list(columns),
+            model_unique_id=model_unique_id,
+        )
+        recommendation = self.RECOMMENDATION.format(model_unique_id=model_unique_id)
+
+        return DBTInsightResult(
+            type=self.TYPE,
+            name=self.NAME,
+            message=failure_message,
+            recommendation=recommendation,
+            reason_to_flag=self.REASON_TO_FLAG,
+            metadata={"columns": columns, "model_unique_id": model_unique_id},
+        )
+
+    def _get_columns_in_model(self, node_id) -> List[str]:
+        if node_id not in self.catalog.get_schema():
+            return []
+        return self.catalog.get_schema()[node_id].keys()
+
+    def _get_columns_with_contract_violation(self, node_id) -> Sequence[str]:
+        columns = []
+        for col in self._get_columns_in_model(node_id):
+            schema = self.catalog.get_schema()[node_id]
+            col_name = col.lower()
+            col_type = schema[col]
+            if any(col_type.lower() == dtype.lower() for dtype in self.dtypes if dtype):
+                if re.match(self.pattern, col_name, re.IGNORECASE) is not None:
+                    columns.append(col_name)
+            elif not re.match(self.pattern, col_name, re.IGNORECASE):
+                columns.append(col_name)
+        return columns
```

Comment:
Not sure if this makes sense. Are we aonly checking if col name matches pattern for specific data type>?

Reply:
```
repos:
- repo: https://github.com/dbt-checkpoint/dbt-checkpoint
 rev: v1.0.0
 hooks:
 - id: check-column-name-contract
   args: [--pattern, "(is|has|do)_.*", --dtypes, boolean text timestamp]
```

so looks like for each data type, they can specify a certain structure. If not specified, applied to all

Reply:
Right now, we are only taking specific pattern, We might not be able to achieve ^. We  need to chnage the config accordingly